codesigndocuitests/codesigngroup.go

```diff
@@ -0,0 +1,409 @@
+package codesigndocuitests
+
+import (
+	"errors"
+	"fmt"
+	"path"
+	"sort"
+	"strings"
+
+	"github.com/bitrise-io/go-utils/colorstring"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/goinp/goinp"
+	"github.com/bitrise-tools/codesigndoc/codesign"
+	"github.com/bitrise-tools/go-xcode/certificateutil"
+	"github.com/bitrise-tools/go-xcode/export"
+	"github.com/bitrise-tools/go-xcode/exportoptions"
+	"github.com/bitrise-tools/go-xcode/plistutil"
+	"github.com/bitrise-tools/go-xcode/profileutil"
+)
+
+// extractCertificatesAndProfiles returns the certificates and provisioning profiles of the given codesign group
+func extractCertificatesAndProfiles(codeSignGroups ...export.CodeSignGroup) ([]certificateutil.CertificateInfoModel, []profileutil.ProvisioningProfileInfoModel) {
+	certificateMap := map[string]certificateutil.CertificateInfoModel{}
+	profilesMap := map[string]profileutil.ProvisioningProfileInfoModel{}
+	for _, group := range codeSignGroups {
```

Comment:
Why put items a map, if returning array?

Reply:
Because of the duplications in the array. 
Sadly Golang doesn't support Set, so the other solution for this, to create a map with a certificate.Serial of with the profile.UDID for key.

Reply:
Got it. Found this example, which could be used in the future:
func unique(intSlice []int) []int {
    keys := make(map[int]bool)
    list := []int{} 
    for _, entry := range intSlice {
        if _, value := keys[entry]; !value {
            keys[entry] = true
            list = append(list, entry)
        }
    }    
    return list
}